scalableminds/webknossos
# Display Loading... instead of No Data when creating a task

This PR changes the behavior of several views. Selection fields that use loaded data now dislpay a loading spinner instead of "No Data" while fetching data.   
Affected select fields are loading projects, users, taskTypes, teams, scripts or datasets

### URL of deployed dev instance (used for testing):
- https://loadingintaskcreationselects.webknossos.xyz

### Steps to test:
- login as a user
- have a big set of data for projects, datasets, scripts and/or task types
- (alternatively you could put something like a sleep method in the fetchData function of view you are testing to simulate the wait for the data )
- visit the view whose fields you want to test
- you should see a loading spinner when trying to select a Project, TaskType, Dataset, Team, User or Script until all data has been fetched (try to reload the page if it loads too fast)
- If there is no data for one of the mentioned fields, it should display "No Data" after the spinner disappears

### Issues:
- fixes #3558

------
- [x] talked about possible refactoring
